src/+replab/+init/Dependency.m

```diff
@@ -0,0 +1,339 @@
+classdef Dependency < replab.Str
+% Describes a RepLAB dependency
+    properties (SetAccess = protected)
+        name % (charstring): Name of the dependency
+        testFilename % (charstring): File to test the existence of
+        inPathFun % (charstring): Function handle that returns true if the library is in the path
+        worksFun % (function_handle): Function handle that returns true if the library is working (e.g. by solving a toy problem), if it throws = false, no argument
+        initFun % (function_handle): Function handle that initializes the library (e.g. adds it to the path), takes a folder name argument
+        githubUsername % (charstring): Name of the GitHub user
+        githubRepository % (charstring): Name of the GitHub repository
+        gitCommit % (charstring): MD5 of the commit
+    end
+
+    methods (Static)
+
+
+        function d = MOcov
+            d = replab.init.Dependency;
+            d.name = 'MOcov';
+            d.inPathFun = @() any(exist('mocov') == 2);
+            d.testFilename = 'MOcov/mocov.m';
+            d.worksFun = @() ~isempty(mocov_get_absolute_path('.'));
+            d.initFun = @(path) addpath(fullfile(path, 'MOcov'));
+            d.githubUsername = 'MOcov';
+            d.githubRepository = 'MOcov';
+            d.gitCommit = '2078bfc5d28a2cd903767ce5f48c81762645ecca';
```

Comment:
We already have that problem. The `replab_easyinstall` script pulls a ZIP file that needs to be updated manually with each dependency change. And we need to update manually a ZIP file in a compound ZIP file! (This motivated the new framework, as the cyclotomics dependency is changed pretty quickly)

Now, updating those dependencies is much easier, as we just need to be sure that those commit IDs are the same as the ones in the modules. We could actually put those IDs as `properties (Constant=true)` so that they float at the top of the `Dependency.m` file and are easy to spot and update.

Reply:
@jdbancal There is a `modules.ini` file in the `external/` folder that contains this information.

Reply:
The release script verifies that commit IDs match.

Reply:
Great! Thanks a lot for looking into that, this greatly improves robustness (and enjoyment ;-) of the code management :+1: